r/src/array_to_vector.cpp

```diff
@@ -693,6 +741,9 @@ std::shared_ptr<Converter> Converter::Make(const std::shared_ptr<DataType>& type
     case Type::BOOL:
       return std::make_shared<arrow::r::Converter_Boolean>(std::move(arrays));
 
+    case Type::BINARY:
```

Comment:
Since this is in your mental cache, we can accept FIXED_BINARY and LARGE_BINARY with minimal effort by using `GetView` instead (and some template dispatching because they don't share the same base class). LARGE_BINARY will require to bound check each value with INT32_MAX.

Reply:
Fixed types: https://issues.apache.org/jira/browse/ARROW-9291